lnrpc/signrpc/driver.go

```diff
@@ -0,0 +1,66 @@
+// +build signerrpc
+
+package signrpc
+
+import (
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+// createNewSubServer is a helper method that will create the new signer sub
+// server given the main config dispatcher method. If we're unable to find the
+// config that is meant for us in the config dispatcher, then we'll exit with
+// an error.
+func createNewSubServer(configRegistry lnrpc.SubServerConfigDispatcher) (lnrpc.SubServer, lnrpc.MacaroonPerms, error) {
+	// We'll attempt to look up the config that we expect, according to our
+	// subServerName name. If we can't find this, then we'll exit with an
+	// error, as we're unable to properly initialize ourselves without this
+	// config.
+	signServerConf, ok := configRegistry.FetchConfig(subServerName)
+	if !ok {
+		return nil, nil, fmt.Errorf("unable to find config for "+
+			"subserver type %s", subServerName)
+	}
+
+	// Now that we've found an object mapping to our service name, we'll
+	// ensure that it's the type we need.
+	config, ok := signServerConf.(*Config)
+	if !ok {
+		return nil, nil, fmt.Errorf("wrong type of config for "+
+			"subserver %s, expected %T got %T", subServerName,
+			&Config{}, signServerConf)
+	}
+
+	// Before we try to make the new signer service instance, we'll perform
+	// some sanity checks on the arguments to ensure taht they're useable.
+	switch {
+	case config.MacService == nil:
+		return nil, nil, fmt.Errorf("MacService must be set to create " +
+			"Signrpc")
```

Comment:
This doesn't allow running with `NoMacaroons`

Reply:
I don't think so. The idea of `NoMacaroons` is that you can use lnd without macaroons? It probably comes back to why we have the `NoMacaroons` flag in the first place. I use it during dev to make it easier to switch between alice, bob and carol on regtest and testnet.

Reply:
Latest version is now usable with no macaroons!